src/Frontend/Page/Delegation.hs

```diff
@@ -33,36 +34,60 @@ import qualified Frontend.Path as U
 
 
 -- | 12. Delegate vote
-data PageDelegateVote = PageDelegateVote (Either Topic Idea) [User]
+data PageDelegateVote = PageDelegateVote (Either Topic Idea) [User] (Maybe (AUID User))
   deriving (Eq, Show, Read)
 
 instance Page PageDelegateVote where isAuthorized = userPage
 
 newtype PageDelegationVotePayload = PageDelegationVotePayload
-    { unPageDelegationVotePayload :: AUID User }
+    { unPageDelegationVotePayload :: Maybe (AUID User) }
   deriving (Eq, Show, Read)
 
 instance FormPage PageDelegateVote where
     type FormPagePayload PageDelegateVote = PageDelegationVotePayload
 
-    formAction (PageDelegateVote scope _users) = case scope of
+    formAction (PageDelegateVote scope _options _mselected) = case scope of
         Left  topic  -> U.delegateVoteOnTopic topic
         Right idea   -> U.delegateVoteOnIdea idea
 
-    redirectOf (PageDelegateVote scope _users) _ = case scope of
+    redirectOf (PageDelegateVote scope _options _mselected) _ = case scope of
         Left  topic  -> U.viewTopic topic
         Right idea   -> U.viewIdea idea
 
-    -- TODO: Show the existing delegation
-    makeForm (PageDelegateVote _scope users) =
-        PageDelegationVotePayload
-        <$> "user-to-delegate" .: DF.choice userList Nothing
+    makeForm (PageDelegateVote _scope options mselected) =
+        PageDelegationVotePayload <$>
+            "selected-delegate" .: DF.validate valid (DF.text (render <$> mselected))
       where
-        userList = (view _Id &&& view (userLogin . unUserLogin . html)) <$> users
-
-    formPage v f p@(PageDelegateVote _scope _users) = semanticDiv p . f $ do
-        -- TODO: Table from users
-        DF.inputSelect "user-to-delegate" v
+        render :: AUID User -> ST
+        render = cs . show . view unAUID
+
+        valid :: ST -> DF.Result (Html ()) (Maybe (AUID User))
```

Comment:
The errors rendered in the `valid` function are shown to the user. The error messages should be translated.